src/FakeItEasy.Analyzer/SymbolExtensions.cs

```diff
@@ -36,5 +37,35 @@ public static string GetFullName(this INamedTypeSymbol type)
 
             return string.Join(".", nameParts);
         }
+
+        /// <summary>
+        /// Returns the name of a type, decorated with its arity if it's a generic type.
+        /// </summary>
+        /// <param name="type">The type for which to get the decorated name.</param>
+        /// <returns>The decorated name of the type.</returns>
+        public static string GetDecoratedName(this INamedTypeSymbol type)

```

Comment:
True, but the other overload *is* used, and this one might be used later. For consistency and future discoverability, I'd rather keep it public, unless it really bothers you.